pkg/oc/bootstrap/docker/host/host.go

```diff
@@ -54,15 +54,39 @@ func (h *HostHelper) CanUseNsenterMounter() (bool, error) {
 	return err == nil && rc == 0, err
 }
 
-func (h *HostHelper) CopyToHost(src, dst string) error {
+// MakeRemoteHostDir creates a specified directory on the remote host filesystem.
+// This is used when Docker runs in VM.
+func (h *HostHelper) MakeRemoteHostDir(dst string) error {
+	_, rc, err := h.runner().Image(h.image).
+		DiscardContainer().
+		Privileged().
+		HostPid().
+		Bind("/:/rootfs:rw").
+		Entrypoint("/bin/bash").
+		Command("-c", fmt.Sprintf("mkdir -p %q", path.Join("/rootfs", dst))).Run()
+	if err != nil {
+		return err
+	}
+	if rc != 0 {
+		return fmt.Errorf("unable to make remote host directory %q: %d", dst, rc)
+	}
+	return nil
+}
+
+// CopyToRemoteHost copies source directory to remote host filesystem.
+// This is used when Docker runs in VM.
+func (h *HostHelper) CopyToRemoteHost(src, dst string) error {
+	if err := h.MakeRemoteHostDir(dst); err != nil {
+		return err
+	}
 	containerID, err := h.runner().
 		Image(h.image).
 		DiscardContainer().
 		Privileged().
 		HostPid().
 		Bind("/:/rootfs:rw").
 		Entrypoint("/bin/bash").
-		Command("-c", fmt.Sprintf("mkdir -p /rootfs/%s && sleep infinity", dst)).Start()
```

Comment:
why didn't this work?  Why would the extra container work instead?

Reply:
@deads2k if the streaming to container is fast it might finish before we make that directory...